app/js/enhancedDate.js

```diff
@@ -1,5 +1,71 @@
+var enhancedDate = (function() {
+	var dateIsSet;
+	var enhancedDateObject = {
+		setDate: setDate,
+		getDate: getDate,
+		getDayName: getDayName,
+		getMonthName: getMonthName,
+		isFuture: isFuture,
+		isToday: isToday
+	};
+	return enhancedDateObject;
+
+
+	function setDate(date) {
+		if (!date) {dateIsSet = new Date(); return;}
+		if (!(date instanceof Date)) {console.warn('sorry, not a date'); return;}
+		dateIsSet = new Date(date);
+	};
+
+	function getDate() {
+		isDateSet();
+		return dateIsSet;
+	}
+
+	function getDayName() {
+		isDateSet();
+		var dayName = dateIsSet.getDay();
+		var dayOfWeekArray = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+		return dayOfWeekArray[dayName];
+	}
+	function isDateSet() {
+		if(!dateIsSet) {console.warn('date hasn\'t been set! Please call setDate first!'); return false;}
+	}
+	function getMonthName() {
```

Comment:
Same deal with this method as with getDayName - better to put this in the module, but outside of this function so the array doesn't get re-created every time this function is called.